metagraph/src/cli/transform_anno_tax.cpp

```diff
@@ -0,0 +1,111 @@
+#include "transform_anno_tax.hpp"
+
+#include <filesystem>
+#include <tsl/hopscotch_set.h>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "annotation/taxonomy/taxonomic_db.hpp"
+#include "common/threads/threading.hpp"
+#include "config/config.hpp"
+#include "cli/load/load_annotation.hpp"
+
+#include "common/logger.hpp"
+
+
+namespace mtg {
+namespace cli {
+
+using mtg::common::logger;
+
+std::vector<std::string> get_anno_filenames(const std::vector<std::string> &paths) {
+    std::vector<std::string> filenames;
+
+    for (const std::string &path: paths) {
+        if (! filesystem::is_directory(path)) {
+            if (!utils::ends_with(path, ".annodbg")) {
+                logger->warn("File '{}' is not an '.annodbg' file. No evaluation for this file.", path);
+                continue;
+            }
+            filenames.push_back(path);
+            continue;
+        }
+        logger->trace("Looking for anno files in directory: '{}'", path);
+        for (auto &rec_file: filesystem::recursive_directory_iterator(path)) {
+            if (utils::ends_with(rec_file.path().string(), ".annodbg")) {
```

Comment:
There are actually nice methods in std::filesystem for checking file extensions.

https://en.cppreference.com/w/cpp/filesystem/path/extension

Reply:
nice, I changed in this way